### ISSUE TITLE
require-default-props error message cannot handle quoted props

### ISSUE THREAD

**Issue by sheepsteak** [yannickcr/eslint-plugin-react#1161]
The following is with `eslint-plugin-react@6.10.3`.

If a propType is quoted for some reason like this:

```js
MyComponent.propTypes = {
  'data-id': PropTypes.string,
};
```
Then the `require-default-props` rule gives the following warning:

```
propType "undefined" is not required, but has no corresponding defaultProp declaration.
```

Howwever, giving the propType a default value like this does satisfy the rule:
```js
MyComponent.defaultProps = {
  'data-id': 'test',
};
```

It seems like this is an issue with the formatting of the error message.

**Comment by ljharb** [comment]
The error is exactly correct - you defined a propType without defining a defaultProp for it.

I agree that the property name showing up as "undefined" in the error message is a bug.

**Comment by Ralf8686** [comment]
Some new? I have same issue with data attribute.
1 - I have error, when defining a defaultProps
2 - I have error react/require-default-props
```
type Props = {
  'data-x'?: string
};
SomeComponent.defaultProps = {
  'data-x': ''
};
```
